odoo/tools/translate.py

```diff
@@ -350,6 +350,45 @@ def translate(cr, name, source_type, lang, source=None):
     res = res_trans and res_trans[0] or False
     return res
 
+
+class LazyStrTranslator(object):
+
+    __slots__ = ['_source']
+    def __init__(self, source):
+        self._source = source
+
+    def __repr__(self):
+        _ = GettextAlias()
```

Comment:
Is it necessary to create a new instance every time? Is there a limitation that prevents using the singleton one?

Reply:
Additionally, I would declare `LazyStrTranslator` below `GettextAlias`, in logical reading order.

Reply:
Additionally, wouldn't it be possible to just reuse `GettextAlias` directly instead of copying `__call__` e.g. by adding a "frame" parameter to `GettextAlias.__call__` such that a frame can be provided and it falls back on getcurrentframe() otherwise?

Reply:
calling it in a common method has the nice side effect than both `LazyStrTranslator` and `GettextAlias` requires to go back 2 frames, no need for an offset